src/preset.ts

```diff
@@ -841,123 +841,84 @@ export async function expandConfigurePreset(folder: string, name: string, worksp
-                const compilerLocation = await execute(whereExecutable, [compilerName], null, {
-                    environment: EnvironmentUtils.create(expandedPreset.environment),
-                    silent: true,
-                    encoding: 'utf8',
-                    shell: true
-                }).result;
-
-                if (!compilerLocation.stdout) {
-                    // Not on PATH, need to set env
-                    const arch = getArchitecture(preset);
-                    const toolset = getToolset(preset);
-
-                    // Get version info for all VS instances.
-                    const vsInstalls = await vsInstallations();
-
-                    // The VS installation to grab developer environment from.
-                    let vsInstall: VSInstallation | undefined;
-
-                    // VS generators starting with Visual Studio 15 2017 support CMAKE_GENERATOR_INSTANCE.
-                    // If supported, we should respect this value when defined. If not defined, we should
-                    // set it to ensure CMake chooses the same VS instance as we use here.
-                    // Note that if the user sets this in a toolchain file we won't know about it,
-                    // which could cause configuration to fail. However the user can workaround this by launching
-                    // vscode from the dev prompt of their desired instance.
-                    // https://cmake.org/cmake/help/latest/variable/CMAKE_GENERATOR_INSTANCE.html
-                    let vsGeneratorVersion: number | undefined;
-                    const matches = preset.generator?.match(/Visual Studio (?<version>\d+)/);
-                    if (matches && matches.groups?.version) {
-                        vsGeneratorVersion = parseInt(matches.groups.version);
-                        const useCMakeGeneratorInstance = !isNaN(vsGeneratorVersion) && vsGeneratorVersion >= 15;
-                        const cmakeGeneratorInstance = getStringValueFromCacheVar(preset.cacheVariables['CMAKE_GENERATOR_INSTANCE']);
-                        if (useCMakeGeneratorInstance && cmakeGeneratorInstance) {
-                            const cmakeGeneratorInstanceNormalized = path.normalize(cmakeGeneratorInstance);
-                            vsInstall = vsInstalls.find((vs) => vs.installationPath
-                                    && path.normalize(vs.installationPath) === cmakeGeneratorInstanceNormalized);
-
-                            if (!vsInstall) {
-                                log.warning(localize('specified.vs.not.found',
-                                    "Configure preset {0}: Visual Studio instance specified by {1} was not found, falling back on default instance lookup behavior.",
-                                    preset.name, `CMAKE_GENERATOR_INSTANCE="${cmakeGeneratorInstance}"`));
-                            }
+            if (compilerName && vscode.workspace.getConfiguration('cmake').get('enableVSDevEnv')) {
```

Comment:
At some point we may want to fully rely on only the `enableVSDevEnv` check, and not depend on what the compiler is, but I think this is good for now since we know that the VS dev env can't provide anything for any other compilers.